File: fixtures/matlab/post_binary_file.m
%% Web Access using Data Import and Export API
uri = 'http://lodstories.isi.edu:3030/american-art/query';
body = fileread('./sample.sparql');
options = weboptions('HeaderFields', {
    'Content-type' 'application/sparql-query'
    'Accept' 'application/sparql-results+json'
});
response = webwrite(uri, body, options);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

header = [
    HeaderField('Content-type', 'application/sparql-query')
    field.AcceptField(MediaType('application/sparql-results+json'))
]';
uri = URI('http://lodstories.isi.edu:3030/american-art/query');
body = FileProvider('./sample.sparql');
response = RequestMessage('post', header, body).send(uri.EncodedURI);

File: fixtures/python/post_binary_file.py
import requests

headers = {
    'Content-type': 'application/sparql-query',
    'Accept': 'application/sparql-results+json',
}

with open('./sample.sparql', 'rb') as f:
    data = f.read().replace(b'\n', b'')

response = requests.post('http://lodstories.isi.edu:3030/american-art/query', headers=headers, data=data)
